presto-cache/src/main/java/com/facebook/presto/cache/alluxio/CacheFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache.alluxio;
+
+import alluxio.conf.AlluxioConfiguration;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import java.net.URI;
+
+public interface CacheFactory
```

Comment:
We might not need this interface or abstraction if this is only for Alluxio. Just call it AlluxioCacheFactory

Reply:
This factory is required for both Hive and Prism. And it produces different filesystems for Hive and Prism. So I think we would need the interface. If we want to call the Interface AlluxioCacheFactory, then we can discuss.
I was thinking if we build another filesystem or memory based caching and decided to use this pattern then this factory would be useful.